modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue

```diff
@@ -0,0 +1,312 @@
+    @esc="cancel" @no-modal="$emit('safe-close')"
+  >
+    <template #secondaryControls>
+      <AposButton
+        type="default" label="Exit"
+        @click="cancel"
+      />
+    </template>
+    <template #primaryControls>
+      <AposButton
+        type="primary" label="Save"
+        :disabled="!valid"
+        @click="submit"
+      />
+    </template>
+    <template #leftRail>
+      <AposModalRail>
+        <div class="apos-modal-array-items">
+          <p v-if="effectiveMin" :class="minError ? 'apos-modal-array-min-error' : ''">
+            Minimum items: {{ effectiveMin }}
+          </p>
+          <p v-if="field.max" :class="maxError ? 'apos-modal-array-max-error' : ''">
+            Maximum items: {{ field.max }}
+          </p>
+          <button :disabled="maxed || itemError" @click.prevent="add">
+            Add Item
+          </button>
+          <ul class="apos-modal-array-items__items">
+            <li
+              class="apos-modal-array-items__item"
+              v-for="( item, index ) in next"
+              :key="item._id"
+            >
+              <button @click.prevent="remove(item._id)">
+                ⓧ
+              </button>
+              <button v-if="index > 0" @click.prevent="up(item._id)">
+                ⬆️
+              </button>
+              <button v-if="index + 1 < next.length" @click.prevent="down(item._id)">
+                ⬇️
+              </button>
+              <button
+                :disabled="itemError"
+                :id="item._id" class="apos-modal-array-items__btn"
+                :aria-selected="item._id === currentId ? true : false"
+                @click="select(item._id)"
+              >
+                {{ label(item) }}
+              </button>
+            </li>
+          </ul>
+        </div>
+      </AposModalRail>
+    </template>
+    <template #main>
+      <AposModalBody>
+        <template #bodyMain>
+          <div class="apos-modal-array-item">
+            <div class="apos-modal-array-item__wrapper">
+              <div class="apos-modal-array-item__pane">
+                <div class="apos-array-item__body">
+                  <AposSchema
+                    v-if="currentId !== false"
+                    :schema="field.schema"
+                    :trigger-validation="triggerValidation"
+                    :utility-rail="false"
+                    :following-values="followingValues()"
+                    v-model="currentDoc"
+                  />
+                </div>
+              </div>
+            </div>
+          </div>
+        </template>
+      </AposModalBody>
+    </template>
+  </AposModal>
+</template>
+
+<script>
+import AposModalParentMixin from 'Modules/@apostrophecms/modal/mixins/AposModalParentMixin';
+import AposEditorMixin from 'Modules/@apostrophecms/modal/mixins/AposEditorMixin';
+import cuid from 'cuid';
+import klona from 'klona';
+import { get } from 'lodash';
```

Comment:
We just need to drop support below Node 14 and we'll get optional chaining to end this nugget.

Reply:
End what nugget? I'm not sure this comment is where you meant to put it.